### PR TITLE
Create a "native gateway" config for DAQ, for the ATA setup (test appliance)

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

**Comment by grafnu** [comment]
Sorry this PR is so long/large -- it just ended up touching a lot of little things.  The main concept that's introduced is that of a "native gateway", which is like the regular DAQ test gateway, except that it deals with a native vlan (rather than a tagged trunk), and is instantiated before actual devices are discovered (rather than on-demand when devices are discovered).

File: cmd/faux [hunk 1]
```diff
@@ -119,8 +119,8 @@ pid=$(docker inspect --format="{{ .State.Pid }}" $cid)
 
 echo DAQ autoclean docker kill $CONTAINER
 
-if [ -n "$realintf" ]; then
-    echo Moving $realintf into $pid...
+if [[ -n $realintf ]]; then
+    echo Moving real $realintf into $pid...
```
Comment (anurag6) [inline cmd/faux:123]: NIT: real interface?

File: daq/container_gateway.py [hunk 2]
```diff
@@ -11,17 +11,16 @@
 
 LOGGER = logger.get_logger('gateway')
 
-
```
Comment (henry54809) [inline daq/container_gateway.py:14]: Why is this line removed? It's standard to have blank 2 lines before class.

File: docker/include/network/scripts/relay_arp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
+
+from scapy.all import sendp,sniff,ARP
```
Comment (henry54809) [inline docker/include/network/scripts/relay_arp:3]: Can we also add this file to bin/check_style.

File: daq/runner.py [hunk 4]
```diff
@@ -296,6 +317,22 @@ def initialize(self):
 
         LOGGER.debug('Done with initialization')
 
+    def _create_stream_monitor(self):
+        return stream_monitor.StreamMonitor(idle_handler=self._handle_system_idle,
+                                            loop_hook=self._loop_hook,
+                                            timeout_sec=20)  # Polling rate
+
+    def _create_native_gateway(self):
+        if not self._native_vlan:
```
Comment (henry54809) [inline daq/runner.py:326]: This function is odd; this will probably be even more convoluted to anyone not familiar with DAQ.  Is there another path that can be used to start a native gateway?

Comment (grafnu) [reply daq/runner.py]: Yes -- I agree that was particularly messy. I've refactored everything, so hopefully it's better now.

File: daq/runner.py [hunk 5]
```diff
@@ -32,6 +32,11 @@
 
 LOGGER = logger.get_logger('runner')
 
+# Totally not valid MAC used for key of a fake device.
+FAKE_DEVICE_MAC = '00:00:00:00:00:00'
```
Comment (henry54809) [inline daq/runner.py:36]: This is not needed now right

**Review by henry54809 — APPROVED** [review]

File: daq/test_modules/ipaddr_module.py [hunk 6]
```diff
@@ -38,14 +37,8 @@ def __init__(self, host, tmpdir, test_name, module_config):
...
         ]
-        self._logger = logger.get_logger('ipaddr_%s' % self.host_name)
-        log_folder = os.path.join(self.tmpdir, 'nodes', self.host_name, 'tmp')
-        os.makedirs(log_folder)
-        log_path = os.path.join(log_folder, 'activate.log')
-        self._file_handler = logging.FileHandler(log_path)
-        formatter = logging.Formatter(_LOG_FORMAT)
-        self._file_handler.setFormatter(formatter)
-        self._logger.addHandler(self._file_handler)
+        log_file = os.path.join(self.tmpdir, 'nodes', self.host_name, 'tmp', 'activate.log')
```
Comment (henry54809) [inline daq/test_modules/ipaddr_module.py:40]: Why is this changed? There’s no need to have a separate custom logging handler

Comment (grafnu) [reply daq/test_modules/ipaddr_module.py:40]: I separated out the logging changes to #865 -- we can address/finalize that first, and then come back to this PR.
